radiation/python/radlw/setcoef_gt4py.py

```diff
@@ -1,220 +1,496 @@
+            #           fraction of the difference (in ln(pressure)) between these
+            #           two values that the layer pressure lies.
+
+            plog = log(pavel)
+            jp = max(1, min(58, 36.0 - 5.0 * (plog + 0.04))) - 1
+            jp1 = jp + 1
+            #  --- ...  limit pressure extrapolation at the top
+            fp = max(0.0, min(1.0, 5.0 * (preflog[0, 0, 0][jp] - plog)))
+
+            #  --- ...  determine, for each reference pressure (jp and jp1), which
+            #           reference temperature (these are different for each
+            #           reference pressure) is nearest the layer temperature but does
+            #           not exceed it. store these indices in jt and jt1, resp.
+            #           store in ft (resp. ft1) the fraction of the way between jt
+            #           (jt1) and the next highest reference temperature that the
+            #           layer temperature falls.
+
+            tem1 = (tavel - tref[0, 0, 0][jp]) / 15.0
+            tem2 = (tavel - tref[0, 0, 0][jp1]) / 15.0
+            jt = max(1, min(4, 3.0 + tem1)) - 1
+            jt1 = max(1, min(4, 3.0 + tem2)) - 1
+            # --- ...  restrict extrapolation ranges by limiting abs(det t) < 37.5 deg
+            ft = max(-0.5, min(1.5, tem1 - (jt - 2)))
+            ft1 = max(-0.5, min(1.5, tem2 - (jt1 - 2)))
+
+            #  --- ...  we have now isolated the layer ln pressure and temperature,
+            #           between two reference pressures and two reference temperatures
+            #           (for each reference pressure).  we multiply the pressure
+            #           fraction fp with the appropriate temperature fractions to get
+            #           the factors that will be needed for the interpolation that yields
+            #           the optical depths (performed in routines taugbn for band n)
+
+            tem1 = 1.0 - fp
+            fac10 = tem1 * ft
+            fac00 = tem1 * (1.0 - ft)
+            fac11 = fp * ft1
+            fac01 = fp * (1.0 - ft1)
+
+            forfac = pavel * stpfac / (tavel * (1.0 + h2ovmr))
+            selffac = h2ovmr * forfac
+
+            #  --- ...  set up factors needed to separately include the minor gases
+            #           in the calculation of absorption coefficient
+
+            scaleminor = pavel / tavel
+            scaleminorn2 = (pavel / tavel) * (colbrd / (coldry + colamt[0, 0, 0][0]))
+
+            tem1 = (tavel - 180.8) / 7.2
+            indminor = min(18, max(1, tem1))
+            minorfrac = tem1 - indminor
+
+            #  --- ...  if the pressure is less than ~100mb, perform a different
+            #           set of species interpolations.
+
+            indfor = indfor
+            forfrac = forfrac
+            indself = indself
+            selffrac = selffrac
+            rfrate = rfrate
+            chi_mls = chi_mls
+
+            if plog > 4.56:
```

Comment:
we probably only want to compare plog to 4.56 once if possible, but why is it > here and >= in the other case? If both comparisons are needed, keep them, but we may want to consider making 4.56 a variable they both use.

Reply:
My mistake, I'll remove the lower one and just define laytrop inside here